sdk/core/Azure.Core/src/Pipeline/AzureOperationScope.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+
+namespace Azure.Core.Pipeline
+{
+    public struct ClientDiagnosticScope: IDisposable
```

Comment:
Does `Client` add anything here?  I'd personally prefer a shorter name like `DiagnosticScope` since we'll be looking at these EVERYWHERE.  😄